sys/include/ztimer/config.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_ztimer
+ * @{
+ *
+ * @file
+ * @brief       ztimer default configuration
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef ZTIMER_CONFIG_H
+#define ZTIMER_CONFIG_H
+
+#include "board.h"
+#include "periph_conf.h"
+
+#include "ztimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * @brief   Use periph_timer as the base timer for ZTIMER_USEC
+ */
+#define CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER (1)
+
+/* for ZTIMER_USEC, use xtimer configuration if available and no ztimer
+ * specific configuration is set. */
+#if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+#  ifndef CONFIG_ZTIMER_USEC_DEV
+#    ifdef XTIMER_DEV
+#      define CONFIG_ZTIMER_USEC_DEV            XTIMER_DEV
+#    endif
+#  endif
+#  ifndef CONFIG_ZTIMER_USEC_BASE_FREQ
+#    ifdef XTIMER_HZ
+#      define CONFIG_ZTIMER_USEC_BASE_FREQ      XTIMER_HZ
+#    endif
+#  endif
+#  ifndef CONFIG_ZTIMER_USEC_WIDTH
+#    ifdef XTIMER_WIDTH
+#      define CONFIG_ZTIMER_USEC_WIDTH          XTIMER_WIDTH
+#    endif
+#  endif
+#endif
+
+/**
+ * @brief   Default timer device for ZTIMER_USEC
+ */
+#ifndef CONFIG_ZTIMER_USEC_DEV
+#define CONFIG_ZTIMER_USEC_DEV          (TIMER_DEV(0))
+#endif
+
+/**
+ * @brief   The frequency of the base ztimer for ZTIMER_USEC
+ */
+#ifndef CONFIG_ZTIMER_USEC_BASE_FREQ
+#define CONFIG_ZTIMER_USEC_BASE_FREQ    (1000000LU)
+#endif
+
+/**
+ * @brief   ZTIMER_USEC optimal minimum value for ztimer_set()
+ *
+ * When scheduling an ISR every timer will be set to max("min", value).
+ *
+ * This is supposed to be defined per-device in e.g., board.h.
+ */
+#ifndef CONFIG_ZTIMER_USEC_MIN
+#define CONFIG_ZTIMER_USEC_MIN          (10)
+#endif
+
+/**
+ * @brief   ZTIMER_USEC base timer width
+ */
+#ifndef CONFIG_ZTIMER_USEC_WIDTH
+#  if (TIMER_0_MAX_VALUE) == 0xFFFF
+#    define CONFIG_ZTIMER_USEC_WIDTH        (16)
+#    define XTIMER_WIDTH (24)
```

Comment:
why is this 24? isn't xtimer_width always 32 when ztimer is used?

Reply:
copy pasta